RoddyCore/src/de/dkfz/roddy/plugins/LibrariesFactory.groovy

```diff
@@ -229,41 +238,64 @@ public class LibrariesFactory extends Initializable {
 
         //Search all plugin folders and also try to join those if possible.
         List<Tuple2<File, String[]>> collectedPluginDirectories = [];
-        def blacklist = [".idea", "out", "Template", ".svn"]
         boolean warningUnzippedDirectoriesMissing = false;
 
         for (File pBaseDirectory : pluginDirectories) {
-	    logger.postSometimesInfo("Parsing plugins folder: ${pBaseDirectory}");
+            logger.postSometimesInfo("Parsing plugins folder: ${pBaseDirectory}");
```

Comment:
Can you split the whole giant loop block into something
``
for (File pBaseDirectory : pluginDirectories) {
   if (isValidPluginDirectory(pBaseDirectory) {
     collectedPluginDirectories << parsePluginDirectory(pBaseDirectory)
  }
}
``

Currently the this is too much code for one function. Consider also splitting parsePluginDirectory() and isValidPluginDirectory() further into parts accessing the filesystem (may be put into RuntimeService?) and parts that are parsing checks.